app/domain/authentication/authn_jwt/jwt_fetch_signing_key_factory.rb

```diff
@@ -0,0 +1,65 @@
+module Authentication
+  module AuthnJwt
+    # Factory that returns the interface implementation of FetchSigningKey
+    class JwtFetchSigningKeyFactory
+      FETCH_SIGNING_KEY_METHODS = {
+        "provider-uri" => JwtFetchProviderUriSigningKey,
+        "jwks-uri" => JwtFetchJwksUriSigningKey
+      }
+
+      def signin_key_implementation(authenticator_input)
+        @authenticator_input = authenticator_input
+
```

Comment:
i think that using the `FETCH_SIGNING_KEY_METHODS` hash adds redundant complexity.
How about using a switch case?
```
case @selected_fetch_key_method
when "provider-uri"
  JwtFetchProviderUriSigningKey.new
when "jwks-uri"
  JwtFetchJwksUriSigningKey.new
else 
  raise("Fetch key method #{@selected_fetch_key_method} does not exist.")
end
```